clients/pkg/promtail/wal/watcher_metrics.go

```diff
@@ -29,6 +31,24 @@ func NewWatcherMetrics(reg prometheus.Registerer) *WatcherMetrics {
 			},
 			[]string{"id"},
 		),
+		droppedWriteNotifications: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "loki",
+				Subsystem: "wal_watcher",
+				Name:      "dropped_write_notifications",
+				Help:      "Number of dropped write notifications due to having one already buffered.",
+			},
+			[]string{"id"},
+		),
+		timerSegmentReads: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: "loki",
+				Subsystem: "wal_watcher",
+				Name:      "read_segment_by_timer",
+				Help:      "Number of segment reads triggered by the backup timer firing.",
+			},
+			[]string{"id"},
+		),
```

Comment:
We should follow the naming conventions for counter and end the metric names in `_total`, what about `loki_wal_watcher_dropped_write_notifications_total` and `loki_wal_watcher_read__total` with a label `reason="timer"`, then if we want to we can track the same metric for `reason="notification"` in the future without having a breaking change due to metric renaming.

Reply:
sorry, lets add the same metric with `reason="notification"` just to avoid confusion as to why it doesn't exist

Reply:
done!